internal/cluster/validations/validations.go

```diff
@@ -312,73 +315,298 @@ func ValidateVipDHCPAllocationWithIPv6(vipDhcpAllocation bool, machineNetworkCID
 	return nil
 }
 
-func ValidateIPAddresses(ipV6Supported bool, obj interface{}) error {
-	var allAddresses []*string
-	var apiVip string
-	var ingressVip string
-	var machineNetworks []*models.MachineNetwork
-	userManagedNetworking := false
-	vipDhcpAllocation := false
+func handleApiVipBackwardsCompatibility(clusterId strfmt.UUID, apiVip string, apiVips []*models.APIVip) ([]*models.APIVip, error) {
```

Comment:
Feel free to ignore the comment - I wonder if `verify` wouldn't be better than `handle` in the name of the function... When I see the word "handle" I think the function will **provide** the compatibility for me, whereas the function here is only checking if my input is obeying the rules

Reply:
The main reason I used `handle` and not `verify` is because that function (and so does `handleIngressVipBackwardsCompatibility`) does not only `verify` the input it gets, but it also constructs a new VIP list, as it does in(the `if apiVip != "" && len(apiVips) == 0` section. So it is actually not just looking at the input, it acts upon it.

WDYT?